### PR TITLE
Fix sign with keypair, and add keychain tests

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: test/keychain.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const test = require('brittle')
+const { swarm } = require('./helpers')
+const Keychain = require('keypear')
+
+test('createServer with keypear + connect', async function (t) {
```
Comment (LuKks) [inline test/keychain.js:5]: It's the same test as in `test/connections.js`, except that in `server.listen()` uses `keys.get()` from keypear instance.

**Comment by LuKks** [comment]
CI Linux passes, although locally I get this:
```
TAP version 13

# createServer with keypear + connect
/self/dht/lib/connect.js:254
    c.encryptedSocket.destroy(attempts ? PEER_CONNECTION_FAILED() : PEER_NOT_FOUND())
                                                                    ^

DHTError: PEER_NOT_FOUND: Peer not found
    at findAndConnect (/self/dht/lib/connect.js:254:69)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async connectAndHolepunch (/self/dht/lib/connect.js:119:3)
Emitted 'error' event on NoiseSecretStream instance at:
    at WritableState.afterDestroy (/self/dht/node_modules/streamx/index.js:444:19)
    at NoiseSecretStream._destroy (/self/dht/node_modules/@hyperswarm/secret-stream/index.js:463:5)
    at WritableState.updateNonPrimary (/self/dht/node_modules/streamx/index.js:189:16)
    at WritableState.update (/self/dht/node_modules/streamx/index.js:174:70)
    at WritableState.afterOpen (/self/dht/node_modules/streamx/index.js:501:27)
    at NoiseSecretStream._predestroy (/self/dht/node_modules/@hyperswarm/secret-stream/index.js:390:7)
    at NoiseSecretStream.destroy (/self/dht/node_modules/streamx/index.js:572:12)
    at findAndConnect (/self/dht/lib/connect.js:254:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async connectAndHolepunch (/self/dht/lib/connect.js:119:3) {
  code: 'PEER_NOT_FOUND'
}
```

File: test/keychain.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  const [a, b] = await swarm(t)
+  const keys = new Keychain()
+  const clientKeyPair = keys.get()
+
+  const server = a.createServer()
+  await server.listen()
+
+  const socket = b.connect(server.publicKey, { keyPair: clientKeyPair })
+  t.is(socket.publicKey, clientKeyPair.publicKey)
```
Comment (LuKks) [inline test/keychain.js:41]: I.e. `t.is` worked here

File: test/keychain.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  const [a, b] = await swarm(t)
+  const keys = new Keychain()
+  const serverKeyPair = keys.get()
+
+  const server = a.createServer()
+  await server.listen(serverKeyPair)
+  t.is(server.publicKey, serverKeyPair.publicKey)
+
+  const socket = b.connect(server.publicKey)
```
[the remark this answers is not in the document]
Comment (LuKks) [reply test/keychain.js:16]: Ah, because it's the bootstrap node

File: test/keychain.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  const clientKeyPair = keys.get()
+
+  const server = a.createServer()
+  await server.listen()
+
+  const socket = b.connect(server.publicKey, { keyPair: clientKeyPair })
+  t.is(socket.publicKey, clientKeyPair.publicKey)
+
+  server.on('connection', (socket) => {
+    t.alike(socket.remotePublicKey, clientKeyPair.publicKey)
```
[the remark this answers is not in the document]
Comment (mafintosh) [reply test/keychain.js:44]: always alike on buffers

**Comment by LuKks** [comment]
@mafintosh Ready!

**Pull request merged**